lib/src/main/java/xyz/gianlu/librespot/core/ApResolver.java

```diff
@@ -72,30 +78,29 @@ public static void refreshPool() throws IOException {
             url.append("type=").append(types[i]);
         }
 
-        HttpURLConnection conn = (HttpURLConnection) new URL(url.toString()).openConnection();
-        conn.connect();
-
-        try (Reader reader = new InputStreamReader(conn.getInputStream())) {
-            JsonObject obj = JsonParser.parseReader(reader).getAsJsonObject();
-            HashMap<String, List<String>> map = new HashMap<>();
-            for (String type : types)
-                map.put(type, getUrls(obj, type));
+        Request request = new Request.Builder()
+                .url(url.toString())
+                .build();
+        Response response = client.newCall(request).execute();
+        if (response.isSuccessful()) {
+            try (Reader reader = response.body().charStream()) {
```

Comment:
Please use a try catch block to make the request. Like this:
https://github.com/librespot-org/librespot-java/blob/31641ce3d87e51dfa8a4cf7ecf1aace46bb441fc/lib/src/main/java/xyz/gianlu/librespot/core/Login5Api.java#L111-L119